### ISSUE TITLE
Validator check for top-level ports in main/federated reactors missing

### ISSUE THREAD

**Issue by Soroosh129** [lf-lang/lingua-franca#618]
User-defined input and output ports in the main/federated reactor are currently allowed.  I don't see any harm in allowing them, but it might be better to disallow them to avoid any confusion.

Nonetheless, the diagram package does not pick them up. Thus, the following program:
```
target C;
main reactor {
    logical action a:int;
    input in:int;
    reaction(in) a {=
        
    =}
    reaction(a) {=
        
    =}
}
```
is rendered as[^1]
![image](https://user-images.githubusercontent.com/4221770/137600031-e18563a6-78c1-4db1-b6e8-bdafa5ddaff1.png)

I think for the sake of a consistent user experience, either we should disallow these ports altogether or render them in the diagram.

[^1]: Note that the issue reported in #607 is also present in this diagram.

**Comment by Soroosh129** [comment]
> If I recall correctly, we had decided not to allow them. 

Then I think a validator check is missing.

**Comment by housengw** [comment]
this issue has been fixed by `7204296a8e01edfb875abfb5e562fcbd3b3f40cf`
